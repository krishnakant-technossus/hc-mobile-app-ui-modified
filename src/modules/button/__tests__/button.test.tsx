// npx jest ./src/modules/button/__tests__/button.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import Button from '../button';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Button', () => {
  describe('Rendering', () => {
    it('renders correctly', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Button testID="button">Primary Button</Button>
        </ThemeProvider>,
      );

      const button = getByTestId('button');

      expect(button).toBeDefined();
    });

    it('renders the label correctly', () => {
      const {getByText} = render(
        <ThemeProvider theme={defaultTheme}>
          <Button testID="button-label">Primary Button</Button>
        </ThemeProvider>,
      );

      const button = getByText('Primary Button');

      expect(button).toBeDefined();
    });

    describe('Variants', () => {
      it('default: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button>Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('primary: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="primary">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('secondary: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="secondary">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('info: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="info">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('success: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="success">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('warning: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="warning">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('danger: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button color="danger">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('outline: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button appearance="outline">Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('small: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Button small>Button</Button>
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('Interactions', () => {
    it('should call the onPress function on click', () => {
      const onPressMock = jest.fn();

      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Button testID="button" onPress={onPressMock}>
            Primary Button
          </Button>
        </ThemeProvider>,
      );

      const button = getByTestId('button');

      fireEvent.press(button);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });

    it('should not call the onPress function on click when disabled', () => {
      const onPressMock = jest.fn();

      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Button testID="button" onPress={onPressMock} disabled>
            Primary Button
          </Button>
        </ThemeProvider>,
      );

      const button = getByTestId('button');

      fireEvent.press(button);

      expect(onPressMock).not.toHaveBeenCalled();
    });
  });
});
