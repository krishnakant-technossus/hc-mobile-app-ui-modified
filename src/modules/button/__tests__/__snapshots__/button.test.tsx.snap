// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering Variants danger: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#EE4335",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="danger"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants default: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00AEFF",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="primary"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants info: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00AEEF",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="info"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants outline: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#00AEFF",
      "borderRadius": 4,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="outline"
    color="primary"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#00AEFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants primary: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00AEFF",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="primary"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants secondary: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1C2530",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="secondary"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants small: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00AEFF",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 32,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="primary"
    disabled={false}
    small={true}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 14,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants success: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#54CD86",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="success"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;

exports[`Button Rendering Variants warning: should match snapshot 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F99E49",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderWidth": 0,
      "display": "flex",
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 0,
      "width": "100%",
    }
  }
  testID="button"
>
  <Text
    appearance="filled"
    color="warning"
    disabled={false}
    small={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
          "fontWeight": "600",
        },
      ]
    }
    testID="button-label"
  >
    Button
  </Text>
</View>
`;
